### ISSUE TITLE
Use `squirrel` for SQL queries

### ISSUE THREAD

**Issue by bond95** [RedHatInsights/insights-results-aggregator#111]
It will be nice to use `squirrel` as layer of abstraction on SQL queries.

> @bond95 @quarckster what about using https://github.com/Masterminds/squirrel? It is not ORM, "just" quite nice abstraction over low level SQL commands.

> _Originally posted by @tisnik in https://github.com/RedHatInsights/insights-results-aggregator/issues/110#issuecomment-589085688_

**Comment by bond95** [comment]
After some research I find out that `squirrel` is not working with dialects, so it's useless, but I found SQL builder that working with dialects and even could be extended https://github.com/doug-martin/goqu

**Comment by natiiix** [comment]
@bond95 I guess we can use it if you want, but I don't see much added value from this. To me, it just seems like a fancy string concatenator.
The `INSERT`/`UPDATE` statements, with the ability to directly pass a Go structure instance to them, look nice, but I don't think it offers similar ticks (automatic conversion into a Go struct) for `SELECT`.

**Comment by bond95** [comment]
@natiiix main reason to use it, is to have different dialect support.

**Comment by Sergey1011010** [comment]
So we end up having ORM(queries builder for different dialects, migrations, mapping structures to queries and vise versa), but our own and built from scratch

**Comment by bond95** [comment]
@Sergey1011010 not exactly, we don't have any other attributes of ORM like associations or hooks, mapping structures we don't use as well (at least for now), we only added query builder and migrations.
